### PR TITLE
Finish implementing License

### PR DESCRIPTION
**Pull request by djmetzle**
Implement the hell out of it.

One weird quirk: The download_license endpoint burps back Base64 not
JSON.

### CODE REVIEW

File: lib/api/license/license.rb [hunk 1]
```diff
@@ -69,4 +69,68 @@ def update_expiration(license_id:, options:)
...
+	end
+
+	def get_license_instance(license_id:, instance_id:)
+		method = "GET"
+		endpoint = "license/" << license_id << "/instance/" << instance_id
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri)
+	end
+
+	def list_tracked_license_instance(license_id:)
```
Comment (g1y) [inline lib/api/license/license.rb]: instances

File: lib/api/license/license.rb [hunk 2]
```diff
@@ -69,4 +69,68 @@ def update_expiration(license_id:, options:)
...
+		method = "GET"
+		endpoint = "license/" << license_id << "/instances"
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri)
+	end
+
+	def list_untracked_license_instance(license_id:)
```
Comment (g1y) [inline lib/api/license/license.rb]: instances

File: lib/api/license/license.rb [hunk 3]
```diff
@@ -69,4 +69,68 @@ def update_expiration(license_id:, options:)
...
+		endpoint = "license/" << license_id << "/instances/untracked"
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri)
+	end
+
+	def revoke_license(license_id:)
+		method = "PUT"
+		endpoint = "license/" << license_id << "/revoke"
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri, options)
```
Comment (g1y) [inline lib/api/license/license.rb]: Options isn't needed here.

File: lib/api/license/license.rb [hunk 4]
```diff
@@ -69,4 +69,68 @@ def update_expiration(license_id:, options:)
...
+	end
+
+	def unarchive_license(license_id:)
+		method = "PUT"
+		endpoint = "license/" << license_id << "/unarchive"
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri, options)
```
Comment (g1y) [inline lib/api/license/license.rb]: Options isn't needed

File: lib/api/license/license.rb [hunk 5]
```diff
@@ -69,4 +69,68 @@ def update_expiration(license_id:, options:)
...
+	end
+
+   # Warning: this returns an RLI in Base64
+	def download_license(license_id:)
+		method = "GET"
+		endpoint = "licensekey/" << license_id
+		uri = ApiUri.build_uri(endpoint)
+		return self.client.request(method, uri, options)
```
Comment (g1y) [inline lib/api/license/license.rb]: same as above

File: lib/api/license/license.rb (hunk 3 above)
Comment (djmetzle) [reply lib/api/license/license.rb]: copy+paste

File: lib/api/license/license.rb (hunk 4 above)
Comment (djmetzle) [reply lib/api/license/license.rb]: copy+paste

File: lib/api/license/license.rb (hunk 5 above)
Comment (djmetzle) [reply lib/api/license/license.rb]: copy+paste

**Comment by djmetzle** [comment]
Fixed?

**Comment by g1y** [comment]
CR :+1: That's the last bit! Nice!

**Pull request merged**